docs/02-app/01-building-your-application/07-configuring/02-eslint.mdx

```diff
@@ -202,6 +202,19 @@ The `next/core-web-vitals` rule set is enabled when `next lint` is run for the f
 
 > The `next/core-web-vitals` entry point is automatically included for new applications built with [Create Next App](/docs/app/api-reference/create-next-app).
 
+### TypeScript
+
+In addition to the Next.js ESLint rules, `create-next-app --typescript` will also add TypeScript-specific lint rules with `next/typescript` to your config:
+
+```json filename=".eslintrc.json"
+{
+  "extends": ["next/core-web-vitals", "next/typescript"]
+}
+```
+
+Those rules are based on [`plugin:@typescript-eslint/recommended`](https://typescript-eslint.io/linting/configs#recommended).
```

Comment:
@JoshuaKGoldberg It'd be also nice as a follow-up to document how one would opt-into type-aware ESLint rules.

Reply:
Yeah! Strong +1.

Since typescript-eslint v8 with the stabilized `projectService` is coming in a few months, my suggestion would be to procrastinate until then. That way we can use the nice:

```diff
{
  "extends": ["next/core-web-vitals", "next/typescript"],
  "parserOptions": {
    "projectService": true
  }
}
```